### PR TITLE
Use implication for skip_active_record/skip_solid

### PR DESCRIPTION
**Pull request by skipkayhil**
[Generator options should default to nil for unset](https://github.com/rails/rails/pull/52802/commits/24c9c4eb4432f2bdd42da7e7a7d579ea7a1909b3)

The generator options are effectively tri-state, with true/false meaning
set by the user and nil being unset. However, some of these options were
defaulting to false, meaning they were being treated as set to false by
the user.

This commit fixes this issue by setting the values to nil by default
instead of false so that they are properly "unset" by user.

---

[Use implication for skip_active_record/skip_solid](https://github.com/rails/rails/pull/52802/commits/766dc27be1d8367f9c8a2c315c67e3d1d416553b)

When skip_solid was added, it used the "old style" option implication
where skip_*? methods delegate to all the options that should be
implied. This has a few downsides such as making it unclear to users
when conflicting options may be specified (for example,
skip_active_record and no_skip_solid).

This commit changes the skip_solid option to use the new style of
implication so that error messages are better and the predicate method
is simpler.

### Checklist

Before submitting the PR make sure the following are checked:

* [ ] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Review by p8 — APPROVED** [review]

**Comment by p8** [comment]
Could skipping activerecord be asserted in the following test?
https://github.com/rails/rails/blob/f698d735bfd237a6e36dc56b781bb42367bac503/railties/test/generators/app_generator_test.rb#L797

**Comment by skipkayhil** [comment]
> Could skipping activerecord be asserted in the following test?

:+1: Added assertion in `test_skip_active_record` from #52808

File: railties/lib/rails/generators/app_base.rb [hunk 1]
```diff
@@ -111,16 +111,16 @@ def self.add_shared_options_for(name)
...
 
-        class_option :skip_solid,          type: :boolean, default: false,
+        class_option :skip_solid,          type: :boolean, default: nil,
                                            desc: "Skip Solid Cache & Queue setup"
 
         class_option :dev,                 type: :boolean, default: nil,
                                            desc: "Set up the #{name} with Gemfile pointing to your Rails checkout"
 
-        class_option :devcontainer,        type: :boolean, default: false,
+        class_option :devcontainer,        type: :boolean, default: nil,
```
Comment (dhh) [inline railties/lib/rails/generators/app_base.rb:123]: What are these false -> nil changes for?

File: railties/lib/rails/generators/app_base.rb [hunk 2]
```diff
@@ -432,7 +432,7 @@ def skip_kamal?
       end
 
       def skip_solid?
-        options[:skip_active_record] || options[:skip_solid]
+        options[:skip_solid]
```
Comment (dhh) [inline railties/lib/rails/generators/app_base.rb:435]: I got this one in already, but the rest of it looks good 👍

File: railties/lib/rails/generators/app_base.rb (hunk 1 above)
Comment (skipkayhil) [reply railties/lib/rails/generators/app_base.rb:123]: Ah, my fault. I separated this change into a separate commit but didn't put the message in the PR description. I've now updated the description with that commit message which explains this part

**Pull request merged**